exercises/nucleotide-count/canonical-data.json

```diff
@@ -39,12 +62,38 @@
           }
         },
         {
-          "description": "strand with invalid nucleotides",
+          "description": "counts all nucleotides correctly in mixed case.",
           "property": "nucleotideCounts",
-          "strand": "AGXXACT",
+          "strand": "aGcTtTtCaTtCtGaCtGcAaCgGgCaAtAtGtCtCtGtGtGgAtTaAaAaAaGaGtGtCtGaTaGcAgC",
           "expected": {
-            "error": "Invalid nucleotide in strand"
+            "A": 20,
+            "C": 12,
+            "G": 17,
+            "T": 21
           }
+        },
+        {
+          "comments" : [
+            "This should call the property/function at least two times and check if it returns the same",
+            "result. This prevents errors where the call aggregates (sums up) the counts of each call."
+          ],
+          "description": "returns the same amounts for consecutive nucleotideCounts calls.",
+          "property": "nucleotideCounts",
+          "strand": "AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGC",
+          "expected" : true
+        },
```

Comment:
This is a case that I actually took over from the concrete JS implementation. I thought it is an interesting thought and it actually might occur.

Reply:
This is a difficult thing to specify in this json file format.
I think it's better left out of the canonical data and implemented at the track level.

Reply:
I agree with you. But how do you make sure it gets implemented in the tracks?

Limiting the possibilities of testing, just to fit into a constrained model or framework is generally a bad idea, though. 
In this case I suggest we should think about a solution to this constraint in the first place (in another discussion of course). In the meantime, as a walkaround, what arguments speak against utilizing descriptions and comments to specify the case? How many tracks use test generators and how many do so blindly?

Reply:
You might want to add a comment at the top of the file, suggesting a check for referential transparency. But I don't think it should have a separate test case.

Reply:
>  how do you make sure it gets implemented in the tracks?

That is not something we need to or should worry about. 

The people maintaining tracks are experienced developers with knowledge of their language, who have spent time reviewing many of the different solutions to exercises, when they identify solutions that are deficient in some way test cases can be added to the track. If these changes are thought to be more generally applicable they can then be added to this general repository.

The test case you are proposing is very implementation dependent and makes no sense in languages with immutable data. So for that reason I do not believe it is necessary to include it in the canonical data.

Reply:
OK, I agree with this point about immutable languages. 
(One could argue now that some cases might be included in the canonical data as suggestions, kind of optional and meant to be mnemonic. Maybe in forms of comments. But this is another topic.)